live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/EventProcessor.java

```diff
@@ -73,22 +76,38 @@ public void process(Exchange exchange)
     {
         boolean allowEvent = repoEventFilterHandler.handleAndGetAllowed(exchange, integrationProperties.alfresco().filter());
         final RepoEvent<DataAttributes<NodeResource>> event = exchange.getIn().getBody(RepoEvent.class);
-        if (allowEvent)
+
+        if (!allowEvent)
         {
-            if (isNotPredictionNodeEvent(event) && isNotPredictionApplyEvent(event))
-            {
-                handleMetadataPropertiesChange(event);
-                handleContentChange(event);
-                handleNodeDeleteEvent(event);
-            }
-            else
-            {
-                log.atDebug().log("Detected prediction event. Further processing of event with id: {} was skipped.", event.getId());
-            }
+            log.atDebug().log("Repository event of id: {} is denied for further processing", event.getId());
+            return;
         }
-        else
+
+        if (isPredictionApplyEvent(event))
         {
-            log.atDebug().log("Repository event of id: {} is denied for further processing", event.getId());
+            log.atDebug().log("Detected prediction apply event. Further processing of event with id: {} was skipped.", event.getId());
+            return;
+        }
+
+        if (isPredictionNodeEvent(event))
+        {
+            handlePredictionNodeEvent(event);
+            return;
+        }
+
+        handleMetadataPropertiesChange(event);
+        handleContentChange(event);
+        handleNodeDeleteEvent(event);
+    }
+
+    private void handlePredictionNodeEvent(RepoEvent<DataAttributes<NodeResource>> event)
+    {
+        ensureThat(isPredictionNodeEvent(event), "Event is not a prediction node event");
```

Comment:
This is fine, but it does seem like we're using the wrong pattern. The calling code literally just made exactly the same check.

Reply:
yep, you're right, I deleted this check